### ISSUE TITLE
GPL vs MIT

### ISSUE THREAD

**Issue by felixguendling** [kahypar/kahypar#174]
Since [mt-kahypar](https://github.com/kahypar/mt-kahypar) has recently been licenced under the MIT license: are there any plans to do the same for this repository, too?

[edit: or at least allow static linking via LGPL?]

**Comment by kittobi1992** [comment]
Hi @felixguendling,

Unfortunately, we can not change the licence of KaHyPar to MIT since we have recently sold a licence.

I am just curious: What prevents you from using Mt-KaHyPar instead of KaHyPar at the moment?

**Comment by felixguendling** [comment]
Mt-KaHyPar references KaHyPar in the [`external_tools`](https://github.com/kahypar/mt-kahypar/tree/master/external_tools) folder which (to my understanding) implies that the same GPL-rules that apply to KaHyPar also apply to Mt-KaHyPar (at least in case of static linking). The GPL "spreads" through all client libraries whereas the MIT license only requires that the copyright and permission notice needs to be included in all copies. I think I'll try to use a precompiled Mt-KaHyPar instead of using the library. Thank you for your quick response!

**Comment by kittobi1992** [comment]
We have changed the license of all header files that we use in Mt-KaHyPar from GPL to MIT. I have to admit that this is currently not easy to see for end users. There are plans to move all these resources that we need in Mt-KaHyPar in a shared repository under MIT license and remove the dependency from KaHyPar.

**Comment by felixguendling** [comment]
Yes, the only thing I can do as a user with reasonable effort is to go through the project dependencies and check which licenses they use in general. Since the `LICENSE` file in KaHyPar is GPL, the basic assumption is that it applies to the whole project. So creating a separate repository that holds the MIT licensed code (currently part of KaHyPar) used by Mt-KaHyPar and KaHyPar would make things easier.

**Comment by kittobi1992** [comment]
That is a good point. I will see if I can setup a repository with shared resources in the next couple of weeks ;)

**Comment by kittobi1992** [comment]
I have removed the dependency to KaHyPar in Mt-KaHyPar:
https://github.com/kahypar/mt-kahypar/pull/153
All files shared between KaHyPar and Mt-KaHyPar are now in a separate repository under MIT license:
https://github.com/kahypar/kahypar-shared-resources

**Comment by felixguendling** [comment]
Wow, that was quick! Thank you so much! :pray:

**Comment by kittobi1992** [comment]
You are welcome :)
